tests/api_fastapi/core_api/routes/public/test_assets.py

```diff
@@ -459,3 +460,32 @@ def test_should_respond_404(self, test_client):
         )
         assert response.status_code == 404
         assert response.json()["detail"] == "The Asset with uri: `s3://bucket/key` was not found"
+
+
+class TestPostAssetEvents(TestAssets):
+    def test_should_respond_200(self, test_client, session):
+        self.create_assets()
+        event_payload = {"asset_uri": "s3://bucket/key/1", "extra": {"foo": "bar"}}
+        response = test_client.post("/public/assets/events", json=event_payload)
+
+        assert response.status_code == 200
+
+        assert response.json() == {
+            "id": mock.ANY,
+            "asset_id": mock.ANY,
+            "uri": "s3://bucket/key/1",
+            "extra": {"foo": "bar", "from_rest_api": True},
+            "source_task_id": mock.ANY,
+            "source_dag_id": mock.ANY,
+            "source_run_id": mock.ANY,
+            "source_map_index": -1,
+            "created_dagruns": [],
+            "timestamp": mock.ANY,
+        }
+
+    def test_invalid_attr_not_allowed(self, test_client, session):
```

Comment:
I think we are missing a test on redacted values. You can check `Connections` and `Variables` to see how this is done.

Also this is done in the legacy tests. I think we should still have such test to ensure that `extra` sensitive values are masked. 

`@pytest.mark.enable_redact`

Reply:
@pierrejeambrun In legacy we have [this](https://github.com/astronomer/airflow/blob/f671fcef549e638d1551c4ad98cea2a2d2794824/tests/api_connexion/endpoints/test_asset_endpoint.py#L612) test for redacted values also this test uses `_check_last_log `. This is the same test which I am facing issue as mentioned in above [comment](https://github.com/apache/airflow/pull/43984#discussion_r1842571385).

Reply:
Interesting, `extra` are redacted in the Log table, but not in the plain response from the API. I think that this is a mistake, querying the events shouldn't return sensitive values in plain text.

I think we need to add the redact part to the `extra` in a similar way as we do for `Connection` and `Variables`, do you want to take that up in a following PR ?